### PR TITLE
Add CONTRIBUTING.md and Update the changelog to reference GitHub releases

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I have added some documentation to the repository and also updated READMEs to refer GitHub releases instead

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  
+Once the snapshot is deployed, you will see a snapshot version in Maven console output.
+Then you can just replace the Plugin POM version in your downstream PRs by this version so that you can demonstrate the build on ci.jenkins.io
+
+## Information for maintainers
+
+### Managing release notes
+
+* Changelog drafts are automatically by link:https://github.com/toolmantim/release-drafter[Release Drafter]
+* See the documentation about Jenkins Release Drafter configuration link:https://github.com/jenkinsci/.github/blob/master/docs/release-drafter.adoc[here]
```
Comment (oleg-nenashev) [inline CONTRIBUTING.adoc]: I have not created a PR for this page yet, in my TODO list

**Comment by jglick** [comment]
> ### [@oleg-nenashev](/oleg-nenashev) [oleg-nenashev](/oleg-nenashev) added the [documentation](/jenkinsci/plugin-pom/labels/documentation)  label [4 hours ago](#event-2377425754)

By the way, now that there is a standard drafter config expecting particular labels, it would be really nice if we could automatically _define_ those labels in all participating repos. Something we can do in the `.github` repo? We would want to let people manually manage labels per repo (I guess), I am just talking about some initial defaults.

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Building
+
+To run and build the repository with integration tests, you can execute
+
+    mvn -Prun-plugin-pom-its clean verify
+
+## Building and using snapshots
+
+Snapshots might be needed to
```
Comment (jglick) [inline CONTRIBUTING.adoc]: …?

File: CONTRIBUTING.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Building and using snapshots
+
+Snapshots might be needed to 
+If you are a plugin maintainer with account on Jenkins artifactory,
+you can deploy a Plugin POM snapshot using this command:
+
+  mvn clean deploy
+  
+Once the snapshot is deployed, you will see a snapshot version in Maven console output.
```
Comment (jglick) [inline CONTRIBUTING.adoc]:
```suggestion
Once the snapshot is deployed, you will see a timestamped snapshot version in Maven console output.
```

**Comment by casz** [comment]
> > ### [@oleg-nenashev](/oleg-nenashev) [oleg-nenashev](/oleg-nenashev) added the [documentation](/jenkinsci/plugin-pom/labels/documentation)  label [4 hours ago](#event-2377425754)
> 
> By the way, now that there is a standard drafter config expecting particular labels, it would be really nice if we could automatically _define_ those labels in all participating repos. Something we can do in the `.github` repo? We would want to let people manually manage labels per repo (I guess), I am just talking about some initial defaults. Also: [isaacs/github#281](https://github.com/isaacs/github/issues/281) / [isaacs/github#371](https://github.com/isaacs/github/issues/371)

When we get access to GitHub Workflows this might work: https://github.com/TimonVS/pr-labeler-action

**Comment by oleg-nenashev** [comment]
I know about label management, but TBH I do not want to touch this topic for now. Maybe later when we have wider adoption of the GitHub-focused flows instead of Confluence and so on

**Comment by oleg-nenashev** [comment]
Fixed comments. Merging so that I can verify dependabot outputs

File: CONTRIBUTING.adoc (hunk 2 above)
Comment (oleg-nenashev) [reply CONTRIBUTING.adoc]: fixed

**Pull request merged**